aws/aws.go

```diff
@@ -0,0 +1,70 @@
+package aws
+
+import (
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+)
+
+// Returns a list of all AWS regions
+func getAllRegions() []string {
+	resolver := endpoints.DefaultResolver()
+	partitions := resolver.(endpoints.EnumPartitions).Partitions()
+
+	var regions []string
+	for _, p := range partitions {
+		for id := range p.Regions() {
+			regions = append(regions, id)
+		}
+	}
+
+	return regions
+}
+
+// GetAllResources - Lists all aws resources
+func GetAllResources() (map[string]map[string][]*string, error) {
```

Comment:
Hm, that's quite a nested return value. Very hard to know what'll be in it! I had to read through all the code to figure out that it's a map of resource type -> map of region ->  resource unique ID. I think.

This is a perfect opportunity to create some custom `struct` types with named parameters. Example:

```go
type AwsAccountResources {
  ResourcesInRegion map[string]AwsRegionResources 
}

type AwsRegionResources struct {
  Instances []Ec2Instance
}

type Ec2Instance struct {
  InstanceId string
}

func GetAllResources() (*AwsAccountResources, error) {

}
```

Now I can read the function signature and types and know exactly what is being returned.